measure/dependent_by_index.go

```diff
@@ -0,0 +1,60 @@
+package measure
+
+import (
+	"encoding/json"
+)
+
+// DependentIndexed is a measure uses a custom cost func to calculate parameter
+// dependent costs for connecting to points by index.
+func DependentIndexed(
+	timeDependent bool,
+	cost func(
+		from,
+		to int,
+		data VehicleData,
+	) float64,
+) DependentByIndex {
+	return &dependentIndexed{
+		cost:          cost,
+		timeDependent: timeDependent,
+	}
+}
+
+// VehicleData holds vehicle specific data, including times by index (ETA, ETD
+// and ETS), a vehicle id, the vehicle's route and the solution value for that
+// vehicle.
+type VehicleData struct {
```

Comment:
You can reorder the struct to save some bytes:
```go
type VehicleData struct {
	VehicleID  string
	Times      Times
	Route      []int
	Index      int
	RouteValue int
}
```

Reply:
done